lib/reversetunnel/srv.go

```diff
@@ -964,10 +936,35 @@ func (s *server) upsertRemoteCluster(conn net.Conn, sshConn *ssh.ServerConn) (*r
 	return site, remoteConn, nil
 }
 
+func (s *server) upsertRemoteClusterWithPeerTunnels(clusterName string, tunnels []types.TunnelConnection) (*remoteSite, error) {
+	s.Lock()
+	defer s.Unlock()
+
+	var site *remoteSite
+	for _, st := range s.remoteSites {
+		if st.GetName() == clusterName {
+			site = st
+			break
+		}
+	}
+	var err error
+	if site == nil {
+		site, err = newRemoteSiteWithPeerTunnels(s, clusterName, tunnels)
```

Comment:
`newRemoteSiteWithPeerTunnels` ends up calling `newRemoteSite` which does a whole bunch of I/O, and we're under a lock here.

Reply:
Still need to address this. I am planning to split out the construction of the remote site from connecting to the remote auth server. That way we can construct the remotesite and store it and connect in the background.